assets/test/components/add_user.spec.js

```diff
@@ -27,11 +27,20 @@ describe('add_user.vue', () => {
   describe('createUser', () => {
     it('redirect to /users', async () => {  
       mock.onPost("/api/users", {user: {name: "Test", username: "testtest"}}).reply(200);
-      await wrapper.setData({name: "Test", username: "testtest"})
+      await wrapper.setData({name: "Test", username: "testtest"});
         
-      await vm.createUser()    
+      await vm.createUser();
 
       expect(window.location.href).toEqual('/users');
     });
+
+    it('show validate error messages', async () => {
+      mock.onPost("/api/users", {user: {name: "", username: ""}}).reply(404);
+      await wrapper.setData({name: "", username: ""});
+
+      await vm.createUser();
+
+      expect(wrapper.element).toMatchSnapshot();
```

Comment:
This is perfectly fine, but for bigger components, you may only want to use snapshot tests to make sure the overall component is rendered correctly, but for testing individual responsibilities, e.g. if the correct error is rendered for the post request, snapshots may be hard to maintain as the test breaks when functionality not related to the specific test changes. For example if I change the text of the button, even though that is unrelated to this test, this test's snapshot will become outdated

You may want to investigate vue test util's `find` https://vue-test-utils.vuejs.org/api/wrapper/find.html#find or `get` https://vue-test-utils.vuejs.org/api/wrapper/#get functions to retrieve the specific tag element and check the error message.